design/storage-class-device-set.md

```diff
@@ -0,0 +1,69 @@
+# Rook StorageClassDeviceSets
+
+**Target version**: Rook 1.1
+
+## Background
+The primary motivation for this feature is to take advantage of the mobility of storage in cloud-based environments by defining storage based on sets of devices consumed as PVCs. In environments like AWS you can request remote storage that can be dynamically provisioned and attached to any node in a given Availability Zone (AZ). The design can also accommodate non-cloud environments through the use of local PVs.
+
+## StorageClassDeviceSet struct
+```go
+// File: pkg/apis/ceph.rook.io/v1/types.go
+package v1
+
+import (
+    v1 "k8s.io/api/core/v1"
+
+    rook "github.com/rook/rook/pkg/apis/rook.io/v1alpha2"
+)
+
+type StorageClassDeviceSets []StorageClassDeviceSet
+
+struct StorageClassDeviceSet {
+    Name                 string // A unique identifier for the set
+    Count                int // Number of devices in this set
+    Resources            v1.ResourceRequirements // Requests/limits for the devices
+    Placement            rook.Placement // Placement constraints for the devices
+    Config               map[string]string // Provider-specific device configuration
+    volumeClaimTemplates []v1.PersistentVolumeClaimSpec // List of PVC templates for the underlying storage devices
+}
+```
+
+A provider will be able to use the StorageClassDeviceSet struct to describe the properties of a particular set of StorageClassDevices. In this design, the notion of a "StorageClassDevice" is an abstract concept, separate from underlying storage devices. In particular, a single StorageClassDevice could be comprised of multiple underlying storage devices, specified by having more than one item in the volumeClaimTemplates field.
+
+A StorageClassDeviceSet will have the following fields associated with it:
+
+* **name**: A name for the set. **[required]**
+* **count**: The number of devices in the set. **[required]**
+* **resources**: The CPU and RAM requests/limits for the devices. Default is no resource requests.
```

Comment:
I don't understand the concept of resources for a device. I understand for a pod, but a device? Can someone clarify that to me? Thanks!

Reply:
The way I understand this from the conversation is that in order to bind a PV to a node may require certain resources. An example of this may be that a very large device may support multiple OSDs, but that implies that the node would need to have the resources capable of supporting such OSDs. This occupied a lot of the discussion, and I believe there is a TODO item to enumerate a covering set of concrete examples that would address your question.

Reply:
Correct. I will be updating the doc with a (hopefully) better explanation and some examples.

Reply:
Ok, makes sense, thanks!